moveit_setup_assistant/CMakeLists.txt

```
@@ -1,169 +1,91 @@
-target_link_libraries(${PROJECT_NAME}_tools
-  ${YAMLCPP_LIBRARIES}
-  ${catkin_LIBRARIES}
-  ${Boost_LIBRARIES}
-  ${QT_LIBRARIES}
+target_include_directories(${PROJECT_NAME}_tools PUBLIC
+  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+  $<INSTALL_INTERFACE:include>
+  ${OMPL_INCLUDE_DIRS}
 )
-
-# Main Widgets Library - all screens (navigation options)
-add_library(${PROJECT_NAME}_widgets
-  src/widgets/start_screen_widget.cpp
-  src/widgets/planning_groups_widget.cpp
-  src/widgets/double_list_widget.cpp
-  src/widgets/kinematic_chain_widget.cpp
-  src/widgets/group_edit_widget.cpp
-  src/widgets/default_collisions_widget.cpp
-  src/widgets/robot_poses_widget.cpp
-  src/widgets/end_effectors_widget.cpp
-  src/widgets/virtual_joints_widget.cpp
-  src/widgets/passive_joints_widget.cpp
-  src/widgets/perception_widget.cpp
-  src/widgets/configuration_files_widget.cpp
-  src/widgets/ros_controllers_widget.cpp
-  src/widgets/controller_edit_widget.cpp
-  src/widgets/navigation_widget.cpp
-  src/widgets/header_widget.cpp
-  src/widgets/setup_assistant_widget.cpp
-  src/widgets/setup_screen_widget.cpp
-  src/widgets/simulation_widget.cpp
-  src/widgets/author_information_widget.cpp
-  ${HEADERS}
+ament_target_dependencies(
+  ${PROJECT_NAME}_tools
+  "moveit_core"
+  "moveit_ros_planning"
+  "moveit_ros_visualization"
+  "rclcpp"
+  "urdf"
+  "srdfdom"
 )
-set_target_properties(${PROJECT_NAME}_widgets PROPERTIES VERSION "${${PROJECT_NAME}_VERSION}")
-target_link_libraries(${PROJECT_NAME}_widgets
+
+add_executable(${PROJECT_NAME}_updater src/collisions_updater.cpp)
+target_include_directories(${PROJECT_NAME}_updater PUBLIC
+  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+  $<INSTALL_INTERFACE:include>)
+target_link_libraries(
+  ${PROJECT_NAME}_updater
   ${PROJECT_NAME}_tools
-  ${QT_LIBRARIES}
-  ${catkin_LIBRARIES}
-  ${Boost_LIBRARIES}
 )
 
-# executables
-add_executable(${PROJECT_NAME} src/setup_assistant_main.cpp)
-target_link_libraries(${PROJECT_NAME}
-  ${PROJECT_NAME}_widgets ${PROJECT_NAME}_tools
-  ${QT_LIBRARIES} ${catkin_LIBRARIES} ${Boost_LIBRARIES} log4cxx
+ament_target_dependencies(
+  ${PROJECT_NAME}_updater
+  "moveit_core"
+  "moveit_ros_planning"
+  "moveit_ros_visualization"
+  "rclcpp"
+  "urdf"
+  "srdfdom"
 )
 
-add_executable(${PROJECT_NAME}_updater src/collisions_updater.cpp)
-target_link_libraries(${PROJECT_NAME}_updater
-  ${PROJECT_NAME}_tools ${catkin_LIBRARIES} ${Boost_LIBRARIES})
-set_target_properties(${PROJECT_NAME}_updater
-                      PROPERTIES OUTPUT_NAME collisions_updater
-                      PREFIX "")
+set_target_properties(
+  ${PROJECT_NAME}_updater
+  PROPERTIES OUTPUT_NAME collisions_updater
+  PREFIX ""
+)
 
-install(TARGETS ${PROJECT_NAME} ${PROJECT_NAME}_updater
-  RUNTIME DESTINATION ${CATKIN_PACKAGE_BIN_DESTINATION})
+install(
+  TARGETS
+    ${PROJECT_NAME}_updater
+  DESTINATION
+    lib/${PROJECT_NAME}
+)
```

```suggestion
install(
  TARGETS
    ${PROJECT_NAME}_tools
    ${PROJECT_NAME}_updater
  EXPORT ${PROJECT_NAME}Targets
  ARCHIVE DESTINATION lib
  LIBRARY DESTINATION lib
  RUNTIME DESTINATION lib/${PROJECT_NAME}
  INCLUDES DESTINATION include
)
install(DIRECTORY launch DESTINATION share/${PROJECT_NAME})
install(DIRECTORY resources DESTINATION share/${PROJECT_NAME})
install(DIRECTORY templates DESTINATION share/${PROJECT_NAME})
ament_export_targets(${PROJECT_NAME}Targets HAS_LIBRARY_TARGET)
ament_export_dependencies(${THIS_PACKAGE_INCLUDE_DEPENDS})
```

from my other review...